kayttooikeus-service/src/main/java/fi/vm/sade/kayttooikeus/service/KayttajatiedotService.java

```diff
@@ -3,12 +3,22 @@
 import fi.vm.sade.kayttooikeus.dto.KayttajatiedotCreateDto;
 import fi.vm.sade.kayttooikeus.dto.KayttajatiedotReadDto;
 import fi.vm.sade.kayttooikeus.dto.KayttajatiedotUpdateDto;
+import fi.vm.sade.kayttooikeus.model.Kayttajatiedot;
 import fi.vm.sade.kayttooikeus.service.LdapSynchronizationService.LdapSynchronizationType;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
 
 public interface KayttajatiedotService {
 
     KayttajatiedotReadDto create(String henkiloOid, KayttajatiedotCreateDto kayttajatiedot, LdapSynchronizationType ldapSynchronization);
 
+    @Transactional
```

Comment:
`Spring recommends that you only annotate concrete classes (and methods of concrete classes) with the @Transactional annotation, as opposed to annotating interfaces`

https://docs.spring.io/spring/docs/4.3.9.RELEASE/spring-framework-reference/html/transaction.html#transaction-declarative-annotations

Eli implementaatioon kuuluvat nämä annotaatiot.

Reply:
Totta. IDEA:n pull member up tehnyt temput.